python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -270,6 +270,48 @@ def compute_child_subset(
                 value=_ValidatedEntitySubsetValue(child_partitions_subset),
             )
 
+    def compute_map_subset(
+        self, to_key: T_EntityKey, subset: EntitySubset[T_EntityKey]
+    ) -> EntitySubset[T_EntityKey]:
+        from_key = subset.key
+        from_partitions_def = self.asset_graph.get(from_key).partitions_def
+        to_partitions_def = self.asset_graph.get(to_key).partitions_def
+
+        if isinstance(to_key, AssetKey):
+            to_subset = self.compute_parent_subset(to_key, subset)
+            if not to_subset.is_empty:
+                return cast(EntitySubset[T_EntityKey], to_subset)
+
+        if isinstance(from_key, AssetKey):
+            to_subset = self.compute_child_subset(to_key, cast(EntitySubset[AssetKey], subset))
+            if not to_subset.is_empty:
+                return to_subset
```

Comment:
I don't think this bit is correct -- the fact that to_key is an AssetKey or from_key is an AssetKey doesn't prove that one is the child of the other.

I would delete these lines and instead do an explicit check. There are three cases we care about:

1. mapping `from` parent `to` child
2. mapping `from` child `to` parent
3. mapping `from` independent `to` independent (i.e. no dependency relationship)

In the first case, we should use `partition_mapping.get_downstream_partitions_for_partitions`, and for the second case we can use `partition_mapping.get_upstream_mapped_partitions_result_for_partitions`. For the final case, we'll also use `partition_mapping.get_upstream_mapped_partitions_result_for_partitions`, even though it's not technically upstream, it's a more flexible function than the downstream version.

So the pseudocode can look like:

```
if from_partitions_def is None or to_partitions_def is None:
    ... # current code

if "from" is a parent of "to":
    to_partitions_subset = partition_mapping.get_downstream_partitions_for_partitions(...)
else:
    to_partitions_subset = partition_mapping.get_upstream_mapped_partitions_result_for_partitions(...).partitions_subset

return EntitySubset(self, key=to_key, ...)
```